pennylane/circuit_drawer.py

```diff
@@ -0,0 +1,916 @@
+        """
+        for idx, target in enumerate(target_list):
+            if np.array_equal(target, target_element):
+                return idx
+
+        target_list.append(target_element)
+
+        return len(target_list) - 1
+
+    def single_parameter_representation(self, par):
+        """Resolve the representation of an Operator's parameter.
+
+        Args:
+            par (Union[qml.variable.Variable, int, float]): The parameter to be rendered
+
+        Returns:
+            str: String representation of the parameter
+        """
+        if isinstance(par, qml.variable.Variable):
+            return par.render(self.show_variable_names)
+
+        return str(round(par, 3))
+
+    @staticmethod
+    def _format_poly_term(coefficient, variable):
+        """Format a term of a polynomial."""
+        if coefficient == 0:
+            return ""
+
+        return " {} {} {}".format(
+            "+" if coefficient > 0 else "-", round(math.fabs(coefficient), 3), variable
+        )
+
+    def operator_representation(self, op, wire):
+        """Resolve the representation of an Operator.
+
+        Args:
+            op (pennylane.operation.Operator): The Operator instance whos representation shall be resolved
+            wire (int): The Operator's wire which is currently resolved
+
+        Returns:
+            str: String representation of the Operator
+        """
+        if isinstance(op, qml.operation.Tensor):
+            constituent_representations = [
+                self.operator_representation(tensor_obs, wire) for tensor_obs in op.obs
+            ]
+
+            return (" " + self.charset.OTIMES + " ").join(constituent_representations)
+
+        name = op.name
+
+        if name in RepresentationResolver.resolution_dict:
+            name = RepresentationResolver.resolution_dict[name]
+
+        if op.name in self.control_wire_dict and wire in [
+            op.wires[control_idx] for control_idx in self.control_wire_dict[op.name]
+        ]:
+            return self.charset.CONTROL
+
+        if op.num_params == 0:
+            return name
+
+        if op.name == "QubitUnitary":
+            mat = op.params[0]
+            idx = RepresentationResolver.index_of_array_or_append(mat, self.unitary_matrix_cache)
+
+            return "U{}".format(idx)
+
+        if op.name == "Hermitian":
+            mat = op.params[0]
+            idx = RepresentationResolver.index_of_array_or_append(mat, self.hermitian_matrix_cache)
+
+            return "H{}".format(idx)
+
+        if op.name == "QuadOperator":
+            par_rep = self.single_parameter_representation(op.params[0])
+
+            return "cos({0})x + sin({0})p".format(par_rep)
+
+        if op.name == "FockStateProjector":
+            n_str = ", ".join([str(n) for n in op.params[0]])
+
+            return (
+                self.charset.PIPE + n_str + self.charset.CROSSED_LINES + n_str + self.charset.PIPE
+            )
+
+        if op.name == "PolyXP":
```

Comment:
It would be nice to create a separate method for this one (it's alone ~100 lines inside of the  ``operator_representation`` method).

Reply:
Yes, I will refactor this part